src/psyclone/psyir/transformations/loop_swap_trans.py

```diff
@@ -144,6 +144,24 @@ def validate(self, node_outer, options=None):
                 "Error in LoopSwap transformation: The inner loop "
                 "has a non-empty symbol table.")
 
+        for node in node_outer.children[0:2]:
```

Comment:
Would be better to use semantic navigation here, e.g. `node_outer.start_expr`, `stop_expr` etc.

Reply:
Agreed